cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -565,106 +663,107 @@ func (scaleSet *ScaleSet) Nodes() ([]cloudprovider.Instance, error) {
 		return scaleSet.instanceCache, nil
 	}
 
-	klog.V(4).Infof("Nodes: starts to get VMSS VMs")
-	splay := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(scaleSet.instancesRefreshJitter + 1)
-	lastRefresh := time.Now().Add(-time.Second * time.Duration(splay))
-
-	orchestrationMode, err := scaleSet.getOrchestrationMode()
+	// Forcefully updating the instanceCache as the instanceCacheSize didn't match curSize or cache is invalid.
+	err = scaleSet.updateInstanceCache()
 	if err != nil {
-		klog.Errorf("failed to get information for VMSS: %s, error: %v", scaleSet.Name, err)
 		return nil, err
 	}
 
-	klog.V(4).Infof("VMSS: orchestration Mode %s", orchestrationMode)
-
-	if orchestrationMode == compute.Uniform {
-		err := scaleSet.buildScaleSetCache(lastRefresh)
-		if err != nil {
-			return nil, err
-		}
-
-	} else if orchestrationMode == compute.Flexible {
-		if scaleSet.manager.config.EnableVmssFlex {
-			err := scaleSet.buildScaleSetCacheForFlex(lastRefresh)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			return nil, fmt.Errorf("vmss - %q with Flexible orchestration detected but 'enableVmssFlex' feature flag is turned off", scaleSet.Name)
-		}
-
-	} else {
-		return nil, fmt.Errorf("Failed to determine orchestration mode for vmss %q", scaleSet.Name)
```

Comment:
This last condition is removed, but shouldn't be an issue.
Otherwise, just delegation to what's in azure_scale_set_instance_cache.go

Reply:
Note that `buildScaleSetCache()` is renamed to `buildScaleSetCacheForUniform()`